File: src/main/java/com/rabbitown/yachat/command/CommandMain.java
package com.rabbitown.yachat.command;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import com.rabbitown.yachat.YaChat;

public class CommandMain implements CommandExecutor {

    YaChat plugin;

    public CommandMain(YaChat plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (command.getName().equals("yachat")) {
            if (args.length == 1) {
                if (args[0].equals("reload")) {
                    onReloadCommand(sender);
                } else {
                    sender.sendMessage("§cUnknown command.");
                }
            } else {
                sender.sendMessage("§cUnknown command.");
            }
        }
        return true;
    }

    void onReloadCommand(CommandSender sender) {
        if (sender.hasPermission("yachat.admin.reload")) {
            sender.sendMessage("§7Reloading plugin...");
            plugin.loadPlugin();
            sender.sendMessage("§aReloaded successfully!");
        } else {
            sender.sendMessage("§cPermission needing: §7yachat.admin.reload");
        }
    }

}

File: src/main/kotlin/com/rabbitown/yachat/command/CommandMain.kt
package com.rabbitown.yachat.command

import com.rabbitown.yachat.YaChat
import com.rabbitown.yalib.module.command.SimpleCommandRemote
import com.rabbitown.yalib.module.command.annotation.Access
import com.rabbitown.yalib.module.command.annotation.Action
import com.rabbitown.yalib.module.locale.YLocale.Companion.sendLocale
import org.bukkit.command.CommandSender

/**
 * @author Yoooooory
 */
class CommandMain : SimpleCommandRemote("yachat", YaChat.instance, listOf("yc", "chat")) {

    @Action("reload")
    @Access(["yachat.command.reload"])
    fun reload(sender: CommandSender) {
        YaChat.instance.reload()
        sender.sendLocale("command.reload.succeed")
    }

}